codegen/templates/http_client.tmpl

```diff
@@ -129,16 +133,24 @@ func (c *{{$clientName}}) {{title .Name}}(
 	switch res.StatusCode {
 		case {{.OKStatusCode.Code}}:
 			var responseBody {{unref .ResponseType}}
+			{{if isPointerType .ResponseType -}}
 			err = res.ReadAndUnmarshalBody(&responseBody)
+			{{else -}}
+			rawBody, err := res.ReadAll()
 			if err != nil {
-				return nil, respHeaders, err
+				return _res, respHeaders, err
+			}
+			err = json.Unmarshal(rawBody, &responseBody)
```

Comment:
line 139-146 seems to duplicate `ReadAndUnmarshalBody`.
Because`.ResponseType` is either a struct/basic type or a pointer to it. Is it simpler to write this piece of code like this:
```
var responseBody {{.ResponseType}}
{{if isPointerType .ResponseType -}}
 			err = res.ReadAndUnmarshalBody(responseBody)	
{{else -}}
             err = res.ReadAndUnmarshalBody(&responseBody)	
{{end}}
...
return responseBody, respHeader, nil
```